src/lib/spinel/CMakeLists.txt

```diff
@@ -26,6 +26,14 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
+add_library(ot-lib-spinel-config INTERFACE)
+
+set(OT_LIB_SPINEL_CONFIG "" CACHE STRING "Lib spinel project-specific config header file")
+if (OT_LIB_SPINEL_CONFIG)
```

Comment:
Here only specify the custom config file for the Spinel module, so we need to add extra custom config file configurations for other lib modules in the future. I think we can unify the config file for all lib modules.

Reply:
Make sense. Any suggestion on the naming for this macro? What about using `OT_LIB_CONFIG`?

Reply:
I updated to make the config for the entire lib. Currently I name it as `OT_LIB_CONFIG`.